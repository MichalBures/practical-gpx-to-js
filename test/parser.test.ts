import * as fs from 'fs';
import * as path from 'path';
import {
	parseGpxString,
	parseGpxMetadata,
	parseGpxWpt,
	parseGpxTrk,
} from '../src/parser';

describe('Parser', () => {
	it('parse .gpx', async () => {
		// let file = fs.readFileSync(path.join(__dirname, 'testTrack.gpx'), {
		let file = fs.readFileSync(path.join(__dirname, 'Praha-88km.gpx'), {
			encoding: 'utf8',
		});

		let result: any = {};

		let parsedGpx: any = await parseGpxString(file);

		let metadata = parseGpxMetadata(parsedGpx);
		result.metadata = metadata;

		let waypoints = parseGpxWpt(parsedGpx);
		result.waypoints = waypoints;

		let tracks = parseGpxTrk(parsedGpx);
		if (tracks !== null) {
			result.trackName = tracks[0].name;
			result.trackLength = tracks[0].trackpoints.length;
			result.tractFirstPoint = tracks[0].trackpoints[0];
			result.tractLastPoint =
				tracks[0].trackpoints[tracks[0].trackpoints.length - 1];
		}

		console.log(result);
		expect(result).toMatchSnapshot();
	});
});
