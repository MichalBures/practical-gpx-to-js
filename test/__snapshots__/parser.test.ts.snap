// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser parse .gpx 1`] = `
Object {
  "metadata": Object {
    "creator": "AllTrails https://www.alltrails.com/ - Praha-8,8km",
    "description": null,
    "name": "Praha-8,8km",
    "time": 2019-10-18T21:50:26.000Z,
  },
  "trackLength": 368,
  "trackName": "Praha-8,8km on AllTrails",
  "tractFirstPoint": Object {
    "altitude": 280,
    "cadence": null,
    "heartRate": null,
    "lat": 50.088387,
    "lon": 14.3919342,
    "speed": null,
    "time": 2010-01-01T00:00:00.000Z,
  },
  "tractLastPoint": Object {
    "altitude": 282,
    "cadence": null,
    "heartRate": null,
    "lat": 50.08826,
    "lon": 14.39154,
    "speed": null,
    "time": 2010-01-01T00:32:55.000Z,
  },
  "waypoints": Array [
    Object {
      "altitude": 280,
      "description": null,
      "lat": 50.088387,
      "lon": 14.3919342,
      "name": "Start",
      "symbol": null,
      "time": 2010-01-01T00:00:00.000Z,
    },
    Object {
      "altitude": 282,
      "description": null,
      "lat": 50.08826,
      "lon": 14.39154,
      "name": "End",
      "symbol": null,
      "time": 2010-01-01T00:32:55.000Z,
    },
  ],
}
`;
